utils/wesnoth-map-diff/src/images.ts

```diff
@@ -36,7 +39,10 @@ const readTerrainImages = async () => {
   const promises = Object
     .entries(dictTerrainType2ImagesPath)
     .map(async ([terrainType, imageName]) => {
-      const image = await Jimp.read(`${imageBasepath}/${imageName}.png`)
+      image = await Jimp.read(`${imageBasepath}/${imageName}.png`)
+      if (!image) {
+        image = await Jimp.read(`../../../images/${imageName}.png`)
```

Comment:
Is it really work?
On the folder `/images` we have no `png` file

Reply:
I dunno, I was just making something up

Reply:
Use the new directory created in [3a92910](https://github.com/wesnoth/wesnoth/pull/7452/commits/3a929108507424b48507b5170727846450e62978) instead? (N.B. I haven't actually put anything there yet...)

Reply:
Yeah. It's a good idea.
Then we can have an image in `utils/wesnoth-map-diff/images/` mapping the terrain string value into an image.
For example, given the terrain:
```
[terrain_type]
    editor_image=impassable-editor
    id=impassable_overlay
    editor_name= _ "Impassable Overlay"
    string=^Xo
    aliasof=_bas,Xt
    mvt_alias=Xt
    editor_group=obstacle, special
    hide_help=yes
[/terrain_type]
```
We should have an image at `utils/wesnoth-map-diff/images/Xo.png`

Reply:
> We should have an image at `utils/wesnoth-map-diff/images/Xo.png`

What should it look like? Just copy an existing one?

Reply:
Yeah. Copying the existing one should work well

Reply:
> Yeah. Copying the existing one should work well

...actually, if we're just going to be using existing images, why not just fix the path so that it finds the original properly, rather than duplicating things?

Reply:
We can do it as well.
There is nothing within the `Xo` `terrain type` object that makes a link from it to an image named `impassable-editor.png`.

We would need to create some mapper into `map-diff` project to map `Xo` into `impassable-editor.png` (and the same for the other tiles that use a different image, if there is one).

It's a good solution, and we can do it, of course. Its maintenance is probably better as well.
But I was thinking about the most simple solution for now.

If you prefer to create the mapper, it's great.

Reply:
I mean... isn't that what the `editor_image=impassable-editor` key is for?

Reply:
I think [b116845](https://github.com/wesnoth/wesnoth/pull/7452/commits/b11684560fd3a5026124be0bad91480874dc4fb3) ought to help?

Reply:
Or, great! I overlooked this property 😅
Maybe we can simplify it a bit by removing the last `if`, since it isn't adding an image path into `acc` (which will be an issue later when calling `Jimp.open`)
```diff
-      if (!acc[terrainType.string.value] && terrainType.string) {
-       acc[terrainType.string.value] = terrainType.string.value
-      }
```

Reply:
well I mean that's only reached if the previous attempts fail; I don't think it'd hurt to keep it as a backup?

Reply:
Well... if you prefer, we can keep it, of course. But I'm unsure if this attempt will be used.
Is there an image filename that matches `terrainType` is the previous ones failed?

Reply:
I'm not sure, but if there isn't, maybe I (or someone) will create one?

Reply:
If there isn't, I think it isn't necessary.
You can instead use a hardcoded filename for a missing image.
For example, [`grid.png`](https://github.com/wesnoth/wesnoth/blob/master/data/core/images/terrain/mask.png) or [`mask.png`](https://github.com/wesnoth/wesnoth/blob/master/data/core/images/terrain/mask.png).
It'll avoid crashing and make us know if there is an image missing.

Reply:
ok, I went with `grid` in [ef9ed61](https://github.com/wesnoth/wesnoth/pull/7452/commits/ef9ed61c5b9d37bc942b2d758acb7bca3b909dd5)

Reply:
Great! Can you run the new `map-diff` with the same diff [from this PR](https://github.com/wesnoth/wesnoth/pull/6057), and check if it's working now?

Also, can you run the tests?
`npm run test`

Reply:
> Great! Can you run the new `map-diff` with the same diff [from this PR](https://github.com/wesnoth/wesnoth/pull/6057), and check if it's working now?
> 

Eh,  I think it'd be easier to just merge this and then see if it passes

> Also, can you run the tests? `npm run test`

Yeah I did; they passed. Marking this as resolved now.